docs/PROTOCOL.md

```diff
@@ -0,0 +1,243 @@
+
+## 2. Terminology
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+- **Channel**: A secure communication pathway between participants
+- **State**: A snapshot of the channel at a point in time
+- **Adjudicator**: A contract that validates state transitions
+- **stateHash**: A cryptographic hash of a channel state
+
+## 3. Data Structures
+
+### 3.1. Channel
+
+```solidity
+struct Channel {
+    address[] participants; // List of participants in the channel
+    address adjudicator;    // Contract that validates state transitions
+    uint64 challenge;       // Duration in seconds for dispute resolution
+    uint64 nonce;           // Unique identifier for channels
+}
+```
+
+### 3.2. State
+
+```solidity
+struct State {
+    bytes data;           // Application-specific data
+    Allocation[] allocations; // Asset allocation for each participant
+    Signature[] sigs;     // Signatures from participants
+}
+```
+
+### 3.3. Allocation
+
+```solidity
+struct Allocation {
+    address destination; // Recipient of funds
+    address token;       // ERC-20 token address
+    uint256 amount;      // Token amount
+}
+```
+
+### 3.4. Signature
+
+```solidity
+struct Signature {
+    uint8 v;
+    bytes32 r;
+    bytes32 s;
+}
+```
+
+### 3.5. Amount
+
+```solidity
+struct Amount {
+    address token;   // ERC-20 token address
+    uint256 amount;  // Token amount
+}
+```
+
+### 3.6. Channel Status
+
+```solidity
+enum Status {
+    VOID,     // Channel not created
+    INITIAL,  // Creation in progress
+    ACTIVE,   // Fully funded and operational
+    DISPUTE,  // Challenge period active
+    FINAL     // Ready to be closed
+}
+```
+
+### 3.7. Protocol Constants
+
+```solidity
+constant uint256 CREATOR = 0; // Creator's participant index
+constant uint256 BROKER = 1;  // Broker's participant index
+constant uint32 CHANOPEN = 7877;  // Magic number for funding state
+constant uint32 CHANCLOSE = 7879; // Magic number for closing state
+```
+
+## 4. Identifier Computation
+
+### 4.1. Channel Identifier
+
+The channelId MUST be computed as:
+
+```solidity
+channelId = keccak256(
+    abi.encode(
+        ch.participants,
+        ch.adjudicator,
+        ch.challenge,
+        ch.nonce
+    )
+);
+```
+
+### 4.2. State Hash
+
+The stateHash MUST be computed as:
+
+```solidity
+stateHash = keccak256(
+    abi.encode(
+        channelId,
+        state.data,
+        state.allocations
+    )
+);
+```
+
+For signature verification, the final hash MUST use Ethereum's personal sign message format:
+```solidity
+signatureHash = keccak256(
+    abi.encodePacked(
+        "\x19Ethereum Signed Message:\n32",
+        stateHash
+    )
+);
+```
+
+## 5. Funding Protocol
+
+### 5.1. Creation Phase
+
+1. The Creator MUST:
```

Comment:
1. Why do we need to distinguish between Creator and non-creator. Does Creator has some unique abilities?
2. If participants collect all signatures to close, why they cannot collect all signatures to open? In my opinion it doesn't change anything, but are there any justifications for it?

Reply:
This simplifies the channel creation flow for the end user, also there is potential application use cases where the user creates a channel and waits for another participant @MaxMoskalenko

Reply:
1. Anyone who propose a funding state is the creator.
2. Yes, we may change this also to incremental closing, we will have to prototype it, maybe it's also painful to collect closing signatures.

I leave the `close()` using consensus for now, in order to save on close() blockchain calls, to avoid having every participant calling close.

Reply:
But still, in nitro the same could have been achieved with participant[0]. And having two methods instead of one with no particular purposes sounds more like complication than simplification. 

+ keep in mind, that this methods shouldn't be "user friendly", since end users will not likely see/use them anyways. That's more for the developers, who could understand (and accept) protocol abstractions